File: leetcodes/191. Number of 1 Bits/numOfOneBits.java
public class Solution {
  // you need to treat n as an unsigned value
  public int hammingWeight(int n) {
    // AND & -> check the last bits if it's 1
    // >>> unsigned right shift, until n is 0
    int ones = 0;
    while(n != 0) {
      ones += (n & 1);
      n = n  >>> 1;
    }
    return ones;
  }
}

File: leetcodes/191. Number of 1 Bits/numOfOneBits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
      ones = 0
      while n:
        n = n & (n - 1)
        ones = ones + 1
        #in java or js we use >>> unsigned right shift (logical right shift 1010 -> 0101; if 1010 >> 1 = 1001 )
        #python has no unsigned right shift
        #n & (n - 1) => able to right shift until no zeros remained
      return ones

'''
class Solution(object):
    def hammingWeight(self, n):
       return bin(n).count('1')
'''